# Eagle Genomics Custom Proteinfer Implementation

This repo is a fork of [https://github.com/google-research/proteinfer](https://github.com/google-research/proteinfer) and provides all the same features, but with two key changes:

## No Pfam Annotation

This tool was intended for use with the [MGnify Protein Database](http://ftp.ebi.ac.uk/pub/databases/metagenomics/peptide_database/current_release/README.txt) which already provides Pfam annotations for each sequence, generated by ProtENN. The Pfam annotations provided by Proteinfer are essentially the same as the ones from ProtENN and so we can save runtime by removing the Pfam inference models from Proteinfer, which has been done here.

## Variable Batch Size

By default, Proteinfer uses a batch size of 64 sequences. Sequences are sorted from longest to shortest before being passed into the network, so that if there's an out-of-memory error, it'll be caught early. However if we're running on GPUs with lots of memory we may be able to use batch sizes larger than this to save time. Since we're removing Pfam inference as well, this saves even more memory and can allow us to use larger batch sizes too.

The default batch size is still 64 but to change it you can run the script as follows:

`python proteinfer.py -i PATH_TO_INPUT_FASTA -o PATH_TO_OUTPUT_TSV --batch_size=BATCH_SIZE`
